### ISSUE TITLE
ActiveRecord::Relation.`scoping` no longer inherit the scope while association resolving

### ISSUE THREAD

**Issue by issei-m** [rails/rails#37090]
### Steps to reproduce

Let's say we have a Category class which supports tree structure enabled by using `acts_as_tree` gem.
So the `categories` table would have a nullable column `parent_id` used to indicate which one is the parent for it.

```ruby
class Category < ApplicationRecord
  acts_as_tree
end
```

And hit the following code in the console:

```
> Category.where.not(id: 1).scoping { Category.find_by(id: 1).present? }
    Category Load (0.6ms)  SELECT "categories".* FROM "categories" WHERE "categories"."id" != $1 AND "categories"."id" = $2 LIMIT $3  [["id", 1], ["id", 1], ["LIMIT", 1]]
  => false
```

That perfectly works well. In fact, the executed SQL is certainly excluding the record having parent_id=1 as you can see.
However, the following doesn't work as expected:

```
> Category.where.not(id: 1).scoping { Category.find_by(parent_id: 1).parent.present? }
    Category Load (1.3ms)  SELECT "categories".* FROM "categories" WHERE "categories"."id" != $1 AND "categories"."parent_id" = $2 LIMIT $3  [["id", 1], ["parent_id", 1], ["LIMIT", 1]]
    Category Load (1.0ms)  SELECT "categories".* FROM "categories" WHERE "categories"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
  => true
```

In the second query, the scope where `parent_id` must be excluded seems to be dropped. So as a result, `true` returned.

From this, I guess the `scoping` functionality is not working well in the queries generated while association resolving.

### Expected behavior

Even in association query, the scope must be kept. And `false` must be returned.

```
> Category.where.not(id: 1).scoping { Category.find_by(parent_id: 1).parent.present? }
    Category Load (1.2ms)  SELECT  "categories".* FROM "categories" WHERE "categories"."id" != $1 AND "categories"."parent_id" = $2 LIMIT $3  [["id", 1], ["parent_id", 1], ["LIMIT", 1]]
    Category Load (0.9ms)  SELECT  "categories".* FROM "categories" WHERE "categories"."id" != $1 AND "categories"."id" = $2 LIMIT $3  [["id", 1], ["id", 1], ["LIMIT", 1]]
  => false
```

Actually, Rails 5.2 was doing so.

### Actual behavior

Already described.

### System configuration
**Rails version**: 6.0

**Ruby version**: 2.6.2 (In Rails 5.2 test, I used the same version of Ruby)

**Comment by issei-m** [comment]
Okay, the change is clearly described in the Changelog, I've overlooked, so I now figured out it's intended behavior change.

Then, in a more further description of our current situation, we're now using `scoping` and `around_action` combination for the soft-deletion of the record (I don't want to argue about the rightness of using soft-deletion here, though) like following:

```ruby
class SomeController < ApplicationController
  around_action :set_scope

  def set_scope
    Category.where(is_deleted: nil).scoping do
      yield
    end
end
```

Then, no deleted category can appear within the current action of the route safely, even though it's loaded by the association but in after Rails 6, we can no longer use of this trick. 
So any idea to realize those things?
Thanks.

@kamipo I'm just pinging you since the behavior changing was apparently proposed by you. (If wrong, I'm sorry)
